### PR TITLE
Update zio, zio-streams, zio-test, ... to 2.0.0

### PR DESCRIPTION
**Pull request by amitksingh1490**
Updates 
* [dev.zio:zio](https://github.com/zio/zio)
* [dev.zio:zio-streams](https://github.com/zio/zio)
* [dev.zio:zio-test](https://github.com/zio/zio)
* [dev.zio:zio-test-sbt](https://github.com/zio/zio)

 from 1.0.15 to 2.0.0.
[GitHub Release Notes](https://github.com/zio/zio/releases/tag/v2.0.0) - [Version Diff](https://github.com/zio/zio/compare/v1.0.15...v2.0.0)

I'll automatically update this PR to resolve conflicts as long as you don't change it yourself.

If you'd like to skip this version, you can just close this PR. If you have any feedback, just mention me in the comments below.

Configure Scala Steward for your repository with a [`.scala-steward.conf`](https://github.com/scala-steward-org/scala-steward/blob/17235ca1e60ff61e35fe7b7cf51163a15ffc435d/docs/repo-specific-configuration.md) file.

Have a fantastic day writing Scala!

<details>
<summary>Ignore future updates</summary>

Add this to your `.scala-steward.conf` file to ignore future updates of this dependency:
```
updates.ignore = [ { groupId = "dev.zio" } ]
```
</details>

labels: library-update, early-semver-major, semver-spec-major, commit-count:1

### CODE REVIEW

**Comment by gchudnov** [comment]
Hi,
wondering if any help is needed to migrate it to the final version of ZIO 2.0 ?

**Comment by gavares** [comment]
I'm happy to help out with this PR if I can.

File: example/src/main/scala/example/AuthenticationServer.scala [hunk 1]
```diff
@@ -4,11 +4,11 @@ import pdi.jwt.{Jwt, JwtAlgorithm, JwtClaim}
 import zhttp.http.Middleware.bearerAuth
 import zhttp.http._
 import zhttp.service.Server
-import zio.{App, ExitCode, URIO}
+import zio.{Clock => _, _}
```
Comment (tusharmath) [inline example/src/main/scala/example/AuthenticationServer.scala]:
```suggestion
import zio._
```

File: zio-http-logging/src/test/scala/zhttp/logging/LogLevelSpec.scala [hunk 2]
```diff
@@ -2,20 +2,20 @@ package zhttp.logging
...
       assertTrue(sorted == exp)
     },
-    testM("encode decode") {
+    test("encode decode") {
       checkAll(Gen.fromIterable(LogLevel.all)) { level =>
         assertTrue(LogLevel.fromString(level.toString) == Some(level))
       }
     },
-    testM("any invalid value should not decode") {
+    test("any value with the exception of defined values for LogLevel should be set to Error log level.") {
```
Comment (tusharmath) [inline zio-http-logging/src/test/scala/zhttp/logging/LogLevelSpec.scala]: description isn't right.

File: zio-http/src/test/scala/zhttp/http/middleware/WebSpec.scala [hunk 3]
```diff
@@ -236,9 +234,9 @@ object WebSpec extends DefaultRunnableSpec with HttpAppTestExtensions {
 
   private def cond(flg: Boolean) = (_: Any) => flg
 
-  private def condM(flg: Boolean) = (_: Any) => UIO(flg)
+  private def condZIO(flg: Boolean) = (_: Any) => ZIO.succeed(flg)
 
-  private def run[R, E](app: HttpApp[R, E]): ZIO[TestClock with R, Option[E], Response] = {
+  private def runApp[R, E](app: HttpApp[R, E]): ZIO[R, Option[E], Response] = {
```
Comment (tusharmath) [inline zio-http/src/test/scala/zhttp/http/middleware/WebSpec.scala:239]: Why is this change req?

File: project/.bloop/zio-http-build.json [hunk 4]
```diff
@@ -0,0 +1,1996 @@
+{
```
Comment (tusharmath) [inline project/.bloop/zio-http-build.json]: Why is this file added?

File: zio-http/src/main/scala/zhttp/http/package.scala [hunk 5]
```diff
@@ -21,4 +21,32 @@ package object http extends PathSyntax with RequestSyntax with RouteDecoderModul
...
+        override def take(implicit trace: Trace): UIO[A] = queue.take.tap(f)
+
+        override def takeAll(implicit trace: Trace): UIO[Chunk[A]] =
+          queue.takeAll.tap(as => ZIO.foreachDiscard(as)(f))
+
+        override def takeUpTo(max: Int)(implicit trace: Trace): UIO[Chunk[A]] =
+          queue.takeUpTo(max).tap(as => ZIO.foreachDiscard(as)(f))
+      }
+    }
+  }
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/http/package.scala]: Will review if there is a better way, to avoid this.

File: zio-http/src/main/scala/zhttp/service/ChannelFuture.scala [hunk 6]
```diff
@@ -16,31 +16,31 @@ final class ChannelFuture[A] private (jFuture: Future[A]) {
...
 }
 
 object ChannelFuture {
-  def make[A](jFuture: => Future[A]): Task[ChannelFuture[A]] = Task(new ChannelFuture(jFuture))
+  def make[A](jFuture: => Future[A]): Task[ChannelFuture[A]] = ZIO.attempt(new ChannelFuture(jFuture))
 
   def unit[A](jFuture: => Future[A]): Task[Unit] = make(jFuture).flatMap(_.execute.unit)
 
-  def asManaged[A](jFuture: => Future[A]): TaskManaged[Unit] = make(jFuture).toManaged_.flatMap(_.toManaged.unit)
+  def asZIO[A](jFuture: => Future[A]): ZIO[Scope, Throwable, Unit] = make(jFuture).flatMap(_.toZIO.unit)
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/ChannelFuture.scala]:
```suggestion
  def scoped[A](jFuture: => Future[A]): ZIO[Scope, Throwable, Unit] = make(jFuture).flatMap(_.toZIO.unit)
```

File: zio-http/src/main/scala/zhttp/service/ChannelFuture.scala [hunk 7]
```diff
@@ -16,31 +16,31 @@ final class ChannelFuture[A] private (jFuture: Future[A]) {
...
         }
         jFuture.addListener(handler)
       })
-      .onInterrupt(UIO(jFuture.removeListener(handler)))
+      .onInterrupt(ZIO.succeed(jFuture.removeListener(handler)))
   }
 
-  def toManaged: ZManaged[Any, Throwable, Option[A]] = {
-    execute.toManaged(_ => cancel(true))
+  def toZIO: ZIO[Scope, Throwable, Option[A]] = {
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/ChannelFuture.scala]:
```suggestion
  def scoped: ZIO[Scope, Throwable, Option[A]] = {
```

File: zio-http/src/main/scala/zhttp/service/Handler.scala [hunk 8]
```diff
@@ -115,26 +115,24 @@ private[zhttp] final case class Handler[R](
...
+                      ZIO.failCause(other)
                   }
               },
             res =>
-              UIO {
-                if (self.isWebSocket(res)) self.upgradeToWebSocket(jReq, res)
-                else resWriter.write(res, jReq)
-              },
+              if (self.isWebSocket(res)) ZIO.succeed(self.upgradeToWebSocket(jReq, res))
+              else ZIO.attempt(resWriter.write(res, jReq)),
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/Handler.scala]: We should wrap the whole block into a `ZIO.attempt`

File: zio-http/src/main/scala/zhttp/http/Http.scala [hunk 9]
```diff
@@ -258,10 +256,10 @@ sealed trait Http[-R, +E, -A, +B] extends (A => ZIO[R, Option[E], B]) { self =>
   final def collect[B1 >: B, C](pf: PartialFunction[B1, C]): Http[R, E, A, C] =
     self >>> Http.collect(pf)
 
-  final def collectManaged[R1 <: R, E1 >: E, A1 <: A, B1 >: B, C](
-    pf: PartialFunction[B1, ZManaged[R1, E1, C]],
+  final def collectScoped[R1 <: R, E1 >: E, A1 <: A, B1 >: B, C](
+    pf: PartialFunction[B1, ZIO[Scope with R1, E1, C]],
   ): Http[R1, E1, A1, C] =
-    self >>> Http.collectManaged(pf)
+    self >>> Http.collectScoped[B1][R1, E1, C](pf)
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/http/Http.scala]: This operator isn't required IMO anymore, `collectZIO` should be enough.

File: zio-http/src/main/scala/zhttp/http/Http.scala [hunk 10]
```diff
@@ -640,6 +627,12 @@ sealed trait Http[-R, +E, -A, +B] extends (A => ZIO[R, Option[E], B]) { self =>
   final def widen[E1, B1](implicit e: E <:< E1, b: B <:< B1): Http[R, E1, A, B1] =
     self.asInstanceOf[Http[R, E1, A, B1]]
 
+  /**
+   * Narrows the type of the input
+   */
+  final def narrow[A1](implicit a: A1 <:< A): Http[R, E, A1, B] =
+    self.asInstanceOf[Http[R, E, A1, B]]
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/http/Http.scala]: unnecessary movement.

File: zio-http/src/main/scala/zhttp/http/Http.scala [hunk 11]
```diff
@@ -735,9 +728,9 @@ object Http {
 
   /**
    * Creates an Http app which accepts a request and produces response from a
-   * managed resource
+   * scoped resource
    */
-  def collectManaged[A]: Http.PartialCollectManaged[A] = Http.PartialCollectManaged(())
+  def collectScoped[A]: Http.PartialCollectScoped[A] = Http.PartialCollectScoped(())
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/http/Http.scala]: Method not required.

**Review by tusharmath — APPROVED** [review]

**Comment by jdsalchow** [comment]
Is there a release with this coming? 😇
